tracer/test/Datadog.Trace.Tests/LambdaHandlerTests.cs

```diff
@@ -18,18 +19,109 @@ public class LambdaHandlerTests
         public void LambdaHandlerGetters()
         {
             LambdaHandler handler = new LambdaHandler("mscorlib::System.Environment::ExpandEnvironmentVariables");
-            handler.GetAssembly().Should().Be("mscorlib");
-            handler.GetFullType().Should().Be("System.Environment");
-            handler.GetMethodName().Should().Be("ExpandEnvironmentVariables");
+            handler.Assembly.Should().Be("mscorlib");
+            handler.FullType.Should().Be("System.Environment");
+            handler.MethodName.Should().Be("ExpandEnvironmentVariables");
         }
 
         [Fact]
-        public void LambdaHandlerBuidParamTypeArray()
+        public void LambdaHandlerParamTypeArray()
         {
             LambdaHandler handler = new LambdaHandler("mscorlib::System.Environment::ExpandEnvironmentVariables");
             handler.ParamTypeArray.Length.Should().Be(2);
             handler.ParamTypeArray[0].Should().Be("System.String");
             handler.ParamTypeArray[1].Should().Be("System.String");
         }
+
+        [Theory]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler::HandlerMethod", "Datadog.Trace.Tests.TestHandler")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler::HiddenBaseMethod", "Datadog.Trace.Tests.TestHandler")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler::OverridenBaseMethod", "Datadog.Trace.Tests.TestHandler")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler2::AbstractBaseMethod", "Datadog.Trace.Tests.TestHandler2")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler::BaseHandlerMethod", "Datadog.Trace.Tests.BaseHandler")]
+        [InlineData("Datadog.Trace.Tests::Datadog.Trace.Tests.TestHandler::NonOverridenBaseMethod", "Datadog.Trace.Tests.BaseHandler")]
```

Comment:
Do you have any planned test cases with the generic handler? I personally wouldn't mind adding code comments of test cases you have in mind right now since you have all the context in your head right now

Reply:
I _did_ and then decided I didn't really want to go to the hassle of supporting generic types/methods, as it requires messing with converting `![]` etc into the right format and there's no request for them right now.

Instead, added a bunch more tests documenting the existing behaviour + limitations. Also added handling for invalid handler names etc.